source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp

```diff
@@ -0,0 +1,318 @@
+#include "MaterialXShadingNodeImpl.h"
+
+#include "MaterialXNode.h"
+#include "Plugin.h"
+#include "Util.h"
+
+#include <MaterialXGenGlsl/GlslShaderGenerator.h>
+#include <MaterialXGenShader/HwShaderGenerator.h>
+#include <MaterialXGenShader/GenContext.h>
+#include <MaterialXGenShader/Util.h>
+#include <MaterialXFormat/XmlIo.h>
+#include <MaterialXRender/StbImageLoader.h>
+
+#include <maya/MFnDependencyNode.h>
+#include <maya/MFragmentManager.h>
+#include <maya/MRenderUtil.h>
+#include <maya/MShaderManager.h>
+#include <maya/MTextureManager.h>
+#include <maya/MPxShadingNodeOverride.h>
+#include <maya/MPxSurfaceShadingNodeOverride.h>
+
+#include <fstream>
+
+namespace
+{
+    // This should be a shared utility
```

Comment:
Minor: coding convention. We don't indent after a namespace declaration. Also should left align arguments in methods.